sdk/keyvault/keyvault-admin/tests.yml

```diff
@@ -15,6 +15,11 @@ stages:
           Path: sdk/keyvault/keyvault-admin/platform-matrix.json
           Selection: sparse
           GenerateVMJobs: true
+      AdditionalMatrixConfigs:
```

Comment:
Both the MatrixConfigs and AdditionalMatrixConfigs reference the same underlying `keyvault-admin/platform-matrix.json` config, so it will just be a duplicated test run?

Reply:
Which I would then expect to fail due to quota since they both deploy an HSM.

Reply:
I noticed azure pipelines failed to run this PR https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1318662&view=results - could this be the reason?

Reply:
Yep! because the matrices are identical we try to generate jobs with duplicate names, which fails devops validation.

Reply:
So you can just remove the whole `AdditionalMatrixConfigs` unless what you intended is to run the full js base suite here as well?

Reply:
Thank you!

Reply:
I removed it here: https://github.com/Azure/azure-sdk-for-js/pull/19956/commits/7c1802ba17e94ac40446670d816ada95bde833d9

Reply:
So what we have right now is not quite right. When I kicked off the build I see: 
![image](https://user-images.githubusercontent.com/753570/150610734-34718b58-9bb2-49fd-8063-f21385f5f11c.png)

But what I would like to see is:
1. one build using the latest service version (where `ServiceVersion` param is undefined) with MHSM
2. one build using service version 7.2 with MHSM

What I see right now is that we only have one build and it runs against the 7.2 service version...

@benbp what I think needs to happen here is something like this in `platform-matrix.json`:

```json
{
  "include": [
    {
      "Agent": {
        "ubuntu-20.04_ManagedHSM": {
          "OSVmImage": "MMSUbuntu20.04",
          "Pool": "azsdk-pool-mms-ubuntu-2004-general",
          "ArmTemplateParameters": "@{ enableHsm = $true }"
        }
      },
      "TestType": "node",
      "NodeTestVersion": "12.x"
    },
    {
      "Agent": {
        "ubuntu-20.04_ManagedHSM": {
          "OSVmImage": "MMSUbuntu20.04",
          "Pool": "azsdk-pool-mms-ubuntu-2004-general",
          "ArmTemplateParameters": "@{ enableHsm = $true }"
        }
      },
      "TestType": "node",
      "NodeTestVersion": "16.x",
      "ServiceVersion": ["7.2"]
    }
  ],
  "displayNames": {
    "7.2": "service_version_7_2"
  }
}

```

Maybe? I haven't looked too closely but what do you think?

 Although it does raise an interesting question about MHSM limitation (which is no longer 1 per region per susbscription, it's now 5 per region per sub) and what we might do when we go past that threshold....

Reply:
That worked! https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1319717&view=results I think

Reply:
Ahh I see what you were trying to do. In that case, you can simplify the matrix config like this, since it does a sparse generation:

```
{
  "include": [
    {
      "Agent": {
        "ubuntu-20.04_ManagedHSM": {
          "OSVmImage": "MMSUbuntu20.04",
          "Pool": "azsdk-pool-mms-ubuntu-2004-general",
          "ArmTemplateParameters": "@{ enableHsm = $true }"
        }
      },
      "TestType": "node",
      "NodeTestVersion": ["12.x", "16.x"],
      "ServiceVersion": ["", "7.2"]
    },
  ],
  "displayNames": {
    "7.2": "service_version_7_2"
  }
}
```

Reply:
Or if you want to be more strict about specifying the version combinations you can use [groupings](https://github.com/Azure/azure-sdk-tools/blob/main/eng/common/scripts/job-matrix/README.md#matrix-config-file-syntax):

```
{
  "include": [
    {
      "Agent": {
        "ubuntu-20.04_ManagedHSM": {
          "OSVmImage": "MMSUbuntu20.04",
          "Pool": "azsdk-pool-mms-ubuntu-2004-general",
          "ArmTemplateParameters": "@{ enableHsm = $true }"
        }
      },
      "Versions": {
          "12.x": {
              "NodeTestVersion": "12.x",
          },
          "16.x_service_version_7_2": {
              "NodeTestVersion": "16.x",
              "ServiceVersion": "7.2"
          }
      },
      "TestType": "node"
    }
  ]
}
```